test/cpp/jit/test_misc.cpp

```diff
@@ -1036,6 +1036,69 @@ void testRecordFunction() {
 
   clearCallbacks();
 
+  // START: thread local / global context check callbacks
+  struct TestContext : public ObserverContext {
+    int a{0};
+    std::string b;
+  };
+  ids.clear();
+  { // START: global test
+    const int test_val = 123;
+    const std::string test_str = "test str";
+    addGlobalCallback(RecordFunctionCallback(
+        [test_val, test_str, &ids](const RecordFunction& /* unused */) {
```

Comment:
TLDR: The changes from the original diff are explicitly added `test_val` to the lambda capture list (total of 4 instances in this PR). 

More detailed explanation: Since `test_val` is a `const int` initialized by an integer literal. As discussed in #44694, the compiler should automatically capture `test_val`. But VS2019 bug flagged this as an error. In fact Clang warns about this in our internal test: 
```
ClangWarning-Wunused-lambda-capture
caffe2/test/cpp/jit/test_misc.cpp:1049:10: warning: lambda capture 'test_val' is not required to be captured for this use
        [test_val, test_str, &ids](const RecordFunction& /* unused */) {
         ^~~~~~~~~
```

Reply:
Thank you. That'll be helpful if people are looking through the PR history later.